main.go

```diff
@@ -401,49 +401,175 @@ func connectAction(ctx *cli.Context) error {
 	return nil
 }
 
-// disconnectAction tries to stop rhscd service, disconnect from Red Hat Insights and finally
+// setupFormatOption ensures the user has supplied a correct `--format` flag
+// and set values in uiSettings, when JSON format is used.
+func setupFormatOption(ctx *cli.Context) error {
+	// This is run after the `app.Before()` has been run,
+	// the uiSettings is already set up for us to modify.
+	format := ctx.String("format")
+	switch format {
+	case "":
+		return nil
+	case "json":
+		uiSettings.isMachineReadable = true
+		uiSettings.isRich = false
+		return nil
+	default:
+		err := fmt.Errorf(
+			"unsupported format: %s (supported formats: %s)",
+			format,
+			`"json"`,
+		)
+		return cli.Exit(err, 1)
+	}
+}
+
+// DisconnectResult is structure holding information about result of
+// disconnect command. The result could be printed in machine-readable format.
+type DisconnectResult struct {
```

Comment:
I did some experiments and it would not help much, because we print progress messages, spinners, etc., when machine readable output is not used. It would be useless to return `DisconnectResult` in such case.

Reply:
OK! Thanks for giving it a try.